src/sagemaker/tensorflow/estimator.py

```diff
@@ -213,16 +212,27 @@ def __init__(self, training_steps=None, evaluation_steps=None, checkpoint_path=N
         if py_version == 'py3' or script_mode:
             if framework_version is None:
                 raise ValueError(EMPTY_FRAMEWORK_VERSION_ERROR)
-
-            if training_steps or evaluation_steps or requirements_file or checkpoint_path:
-                raise ValueError(
-                    '{} are deprecated in script mode. Please do not set these arguments.'
-                    .format(', '.join(self._DEPRECATED_ARGS))
-                )
+            self._check_framework_args(training_steps, evaluation_steps, requirements_file, checkpoint_path)
```

Comment:
seems like ll. 212-215 could be wrapped up in a more general `_check_args` function.